templates/common/infra/bicep/core/database/postgresql/flexibleserver.bicep

```diff
@@ -0,0 +1,42 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+param sku object
+param storage object
+param administratorLogin string
+@secure()
+param administratorLoginPassword string
+param databaseNames array = []
+param enableFirewall bool = false
+
+// PostgreSQL version
+param version string
+
+resource postgresServer 'Microsoft.DBforPostgreSQL/flexibleServers@2022-01-20-preview' = {
+  location: location
+  tags: tags
+  name: name
+  sku: sku
+  properties: {
+    version: version
+    administratorLogin: administratorLogin
+    administratorLoginPassword: administratorLoginPassword
+    storage: storage
+    highAvailability: {
+      mode: 'Disabled'
+    }
+  }
+
+  resource database 'databases' = [for name in databaseNames: {
+    name: name
+  }]
+
+  resource firewall 'firewallRules' = if (enableFirewall) {
+    name: 'postgresql-firewall'
+    properties: {
+        startIpAddress: '0.0.0.0'
+        endIpAddress: '255.255.255.255'
```

Comment:
I had '0.0.0.0' in my samples, for just Azure IPs, but all the other ones use '255.255.255.255', with a comment about including the local machine. That seems strange to me since that seems like it's allowing all IPs, but maybe I'm missing something?

See here for mention of 0.0.0.0 - 0.0.0.0:
https://learn.microsoft.com/en-us/azure/templates/microsoft.sql/servers/firewallrules?pivots=deployment-language-bicep#serverfirewallruleproperties

Reply:
I could add booleans for 'allowAzureIPsFirewall', 'allowAllIPsFirewall', as well as an array 'allowedSingleIPs' for whitelisting individual IPs. That would cover everything I've seen so far. I'm just flummoxed as to why it'd be okay to allow all IPs.

Reply:
@pamelafox The main reason why they were open to the internet was to allow a local dev server to connect to the DB hosted in Azure.

We want to change this behavior to be behind a flag, something like `secureMode` true or false, with `true` being a truly private VNet for securing DB in the near future.

I am in favor of the options 'allowAzureIPsFirewall', 'allowAllIPsFirewall', 'allowedSingleIPs' options that you've proposed for full flexibility.

Reply:
Okay, got it, I'll add the three options and test them out in my samples.

Reply:
I've added the discussed options and tested setting each of them in two different samples.
https://github.com/pamelafox/flask-surveys-container-app
https://github.com/Azure-Samples/bindings-dapr-nodejs-cron-postgres/compare/main...pamelafox:bindings-dapr-nodejs-cron-postgres:postgres-update?expand=1

I checked Networking tab of servers after deploying, looked good.

<img width="1220" alt="Screenshot 2023-01-30 at 4 22 45 PM" src="https://user-images.githubusercontent.com/297042/215626798-6437252a-a734-427c-91fd-a9f281746d4e.png">

<img width="1290" alt="Screenshot 2023-01-30 at 4 15 50 PM" src="https://user-images.githubusercontent.com/297042/215626504-bdb5333d-fcb2-49a7-b56c-25cd5f610878.png">

Reply:
Thanks!